app/controllers/users_controller.rb

```diff
@@ -1,24 +1,25 @@
 class UsersController < ApplicationController
   before_action :require_login, only: %w[edit update destroy]
   before_action :set_user, only: %w[show]
-  http_basic_authenticate_with name: 'intern', password: ENV['INTERN_PASSWORD'] || 'test'
+  http_basic_authenticate_with name: 'intern', password: ENV['INTERN_PASSWORD'] || 'test', only: %i(new create)
 
   def index
     @categories = Category.order('position')
-    @users = User.in_school.not_advisers.order('updated_at desc')
+    @users = User.order(updated_at: :desc)
+    @target = params[:target] || 'all'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.